exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java

```diff
@@ -375,6 +378,30 @@ protected void cleanup() {
     public FragmentExecutor getFragmentRunner(final FragmentHandle handle) {
       return runningFragments.get(handle);
     }
+
+    public void receiveRuntimeFilter(final RuntimeFilterWritable runtimeFilter) {
+      BitData.RuntimeFilterBDef runtimeFilterDef = runtimeFilter.getRuntimeFilterBDef();
+      boolean toForeman = runtimeFilterDef.getToForeman();
+      QueryId queryId = runtimeFilterDef.getQueryId();
+      //to foreman
+      if (toForeman) {
+        Foreman foreman = queries.get(queryId);
+        if (foreman != null) {
+          foreman.getRuntimeFilterManager().registerRuntimeFilter(runtimeFilter);
```

Comment:
the `registerRuntimeFilter` and eventual broadcast all is happening on rpc thread which is not good since Rpc thread should be released ASAP. Please follow a model similar to **cancelForeman** in WorkManager to submit the `registerRuntimeFilter` task as new Runnable to WorkManager executor